mongomock/collection.py

```diff
@@ -1649,6 +1649,18 @@ def _extend_collection(out_collection, field, expression):
                 if k == '$match':
                     out_collection = [doc for doc in out_collection
                                       if filter_applies(v, doc)]
+                elif k == '$lookup':
```

Comment:
For such a big new feature, we require that you add an end to end test to make sure that `pymongo` behaves like `mongomock`: add something in `test__mongomock.py` please.

Reply:
Since the testing strategy with `MultiCollection` only allows for testing against one collection at a time without major overhaul or hacking, I decided to run the `lookup` against the same collection.

Reply:
Sounds good to me. Thanks for adding.